src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -214,6 +214,12 @@ private[snowflake] class JDBCWrapper {
   private def isQuoted(name: String): Boolean = {
     name.startsWith("\"") && name.endsWith("\"") 
   }
+  private def quotedName(name: String): String = {
+      if (name.matches("[_A-Z]([_0-9A-Z])*"))
+        "\"" + name.toUpperCase + "\""
```

Comment:
Totally. Thank you so much for the thorough explanation. The regex pattern now includes lowercase letters as well, which would ensure identifier legality.